File: generate_icons.py
#!/usr/bin/env python3
"""
Generate app icons for Whads PWA
Requires: pip install pillow
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size, filename):
    """Create a WhatsApp-style icon with gradient background"""
    
    # Create image with transparent background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Draw rounded rectangle with gradient effect
    # Create gradient from #25D366 to #128C7E
    for y in range(size):
        # Calculate color for this row
        ratio = y / size
        r = int(37 + (18 - 37) * ratio)
        g = int(211 + (140 - 211) * ratio)
        b = int(102 + (126 - 102) * ratio)
        color = (r, g, b, 255)
        
        # Draw horizontal line
        draw.line([(0, y), (size, y)], fill=color, width=1)
    
    # Create rounded corners mask
    mask = Image.new('L', (size, size), 0)
    mask_draw = ImageDraw.Draw(mask)
    radius = int(size * 0.2)
    mask_draw.rounded_rectangle([(0, 0), (size, size)], radius=radius, fill=255)
    
    # Apply mask
    img.putalpha(mask)
    
    # Draw WhatsApp "W" letter
    try:
        # Try to use a bold font
        font_size = int(size * 0.5)
        try:
            font = ImageFont.truetype("arialbd.ttf", font_size)
        except:
            try:
                font = ImageFont.truetype("Arial Bold.ttf", font_size)
            except:
                font = ImageFont.load_default()
    except:
        font = ImageFont.load_default()
    
    # Draw white "W" in center
    text = "W"
    
    # Get text bounding box
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    # Calculate position to center text
    x = (size - text_width) // 2
    y = (size - text_height) // 2 - int(size * 0.05)
    
    # Draw text with white color
    draw.text((x, y), text, fill=(255, 255, 255, 255), font=font)
    
    # Draw a chat bubble outline
    bubble_size = int(size * 0.6)
    bubble_x = (size - bubble_size) // 2
    bubble_y = (size - bubble_size) // 2
    
    # Save image
    img.save(filename, 'PNG')
    print(f"✓ Created {filename} ({size}x{size})")

def main():
    """Generate all required icons"""
    
    # Create icons directory if it doesn't exist
    icons_dir = 'icons'
    if not os.path.exists(icons_dir):
        os.makedirs(icons_dir)
    
    print("🎨 Generating Whads PWA Icons...")
    print()
    
    # Generate icons
    create_icon(192, os.path.join(icons_dir, 'icon-192.png'))
    create_icon(512, os.path.join(icons_dir, 'icon-512.png'))
    
    print()
    print("✅ All icons generated successfully!")
    print(f"📁 Icons saved to: {os.path.abspath(icons_dir)}")

if __name__ == '__main__':
    try:
        main()
    except ImportError:
        print("❌ Error: Pillow library not found")
        print("📦 Install it with: pip install pillow")
        print()
        print("Alternative: Open generate-icons.html in a browser to download icons manually")

# Made with Bob

File: generate-icons.js
const fs = require('fs');
const { createCanvas } = require('canvas');

function generateIcon(size, filename) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Create gradient background
    const gradient = ctx.createLinearGradient(0, 0, size, size);
    gradient.addColorStop(0, '#25D366');
    gradient.addColorStop(1, '#128C7E');
    
    // Draw rounded rectangle background
    const radius = size * 0.2;
    ctx.fillStyle = gradient;
    ctx.beginPath();
    ctx.moveTo(radius, 0);
    ctx.lineTo(size - radius, 0);
    ctx.quadraticCurveTo(size, 0, size, radius);
    ctx.lineTo(size, size - radius);
    ctx.quadraticCurveTo(size, size, size - radius, size);
    ctx.lineTo(radius, size);
    ctx.quadraticCurveTo(0, size, 0, size - radius);
    ctx.lineTo(0, radius);
    ctx.quadraticCurveTo(0, 0, radius, 0);
    ctx.closePath();
    ctx.fill();
    
    const centerX = size / 2;
    
    // Draw large chat bubble
    const bubbleSize = size * 0.35;
    const bubbleY = size * 0.35;
    
    // White circle for bubble background
    ctx.fillStyle = 'white';
    ctx.beginPath();
    ctx.arc(centerX, bubbleY, bubbleSize, 0, Math.PI * 2);
    ctx.fill();
    
    // Draw chat bubble shape inside
    ctx.strokeStyle = '#25D366';
    ctx.lineWidth = size * 0.025;
    ctx.lineCap = 'round';
    ctx.lineJoin = 'round';
    
    // Main bubble rounded rectangle
    const bubbleInnerSize = bubbleSize * 0.65;
    const bubbleLeft = centerX - bubbleInnerSize / 2;
    const bubbleTop = bubbleY - bubbleInnerSize / 2;
    const bubbleRadius = bubbleInnerSize * 0.2;
    
    ctx.beginPath();
    ctx.moveTo(bubbleLeft + bubbleRadius, bubbleTop);
    ctx.lineTo(bubbleLeft + bubbleInnerSize - bubbleRadius, bubbleTop);
    ctx.quadraticCurveTo(bubbleLeft + bubbleInnerSize, bubbleTop, bubbleLeft + bubbleInnerSize, bubbleTop + bubbleRadius);
    ctx.lineTo(bubbleLeft + bubbleInnerSize, bubbleTop + bubbleInnerSize - bubbleRadius);
    ctx.quadraticCurveTo(bubbleLeft + bubbleInnerSize, bubbleTop + bubbleInnerSize, bubbleLeft + bubbleInnerSize - bubbleRadius, bubbleTop + bubbleInnerSize);
    ctx.lineTo(bubbleLeft + bubbleRadius, bubbleTop + bubbleInnerSize);
    ctx.quadraticCurveTo(bubbleLeft, bubbleTop + bubbleInnerSize, bubbleLeft, bubbleTop + bubbleInnerSize - bubbleRadius);
    ctx.lineTo(bubbleLeft, bubbleTop + bubbleRadius);
    ctx.quadraticCurveTo(bubbleLeft, bubbleTop, bubbleLeft + bubbleRadius, bubbleTop);
    ctx.stroke();
    
    // Draw bubble tail
    ctx.beginPath();
    ctx.moveTo(bubbleLeft + bubbleInnerSize * 0.2, bubbleTop + bubbleInnerSize);
    ctx.lineTo(bubbleLeft + bubbleInnerSize * 0.15, bubbleTop + bubbleInnerSize + bubbleInnerSize * 0.15);
    ctx.lineTo(bubbleLeft + bubbleInnerSize * 0.35, bubbleTop + bubbleInnerSize);
    ctx.stroke();
    
    // Draw three dots inside bubble
    const dotRadius = bubbleInnerSize * 0.08;
    const dotY = bubbleY;
    const dotSpacing = bubbleInnerSize * 0.25;
    
    ctx.fillStyle = '#25D366';
    
    // Left dot
    ctx.beginPath();
    ctx.arc(centerX - dotSpacing, dotY, dotRadius, 0, Math.PI * 2);
    ctx.fill();
    
    // Center dot
    ctx.beginPath();
    ctx.arc(centerX, dotY, dotRadius, 0, Math.PI * 2);
    ctx.fill();
    
    // Right dot
    ctx.beginPath();
    ctx.arc(centerX + dotSpacing, dotY, dotRadius, 0, Math.PI * 2);
    ctx.fill();
    
    // Draw "Whads!" text below bubble
    const textY = size * 0.75;
    const fontSize = size * 0.18;
    
    ctx.fillStyle = 'white';
    ctx.font = `bold ${fontSize}px Arial, sans-serif`;
    ctx.textAlign = 'center';
    ctx.textBaseline = 'middle';
    
    // Add text shadow for better visibility
    ctx.shadowColor = 'rgba(0, 0, 0, 0.3)';
    ctx.shadowBlur = size * 0.015;
    ctx.shadowOffsetX = 0;
    ctx.shadowOffsetY = size * 0.01;
    
    ctx.fillText('Whads!', centerX, textY);
    
    // Save to file
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(filename, buffer);
    console.log(`✅ Generated: ${filename}`);
}

// Generate all icons
console.log('🎨 Generating Whads! icons...\n');

// Create icons directory if it doesn't exist
if (!fs.existsSync('icons')) {
    fs.mkdirSync('icons');
}

generateIcon(32, 'favicon.png');
generateIcon(192, 'icons/icon-192.png');
generateIcon(512, 'icons/icon-512.png');

console.log('\n✨ All icons generated successfully!');

// Made with Bob
